dev/release/post-13-msys2.sh

```diff
@@ -45,10 +45,13 @@ if ! git remote | grep -q '^upstream$'; then
   exit 1
 fi
 
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "${SOURCE_DIR}/git-vars.sh"
+
 echo "Updating repository: ${repository}"
 git fetch --all --prune --tags --force -j$(nproc)
-git checkout master
-git rebase upstream/master
+git checkout ${DEFAULT_BRANCH}
+git rebase upstream/${DEFAULT_BRANCH}
```

Comment:
We should not change this because the repository is the a of msys2/MINGW-packages not apache/arrow.

Reply:
Thank you for this feedback, @kou. 

I initially included the change in this file to use the default branch of the msys2/MINGW-packages repository, to make the transition smoother if they change the name of their default branch. My thought was that after the `cd` call on line 35, sourcing `git-vars.sh` would get the default branch value of the msys2/MINGW-packages repository. However, I have realized that in order for the `SOURCE_DIR` variable to be computed correctly, this script would need to be executed with a full path.

I will remove the change to use the `git-vars.sh` in this file and update the pull request description.

Reply:
I have also reverted the changes to `post-04-website.sh` and `post-09-docs.sh`, as they also reference another repository, `apache/arrow-site`.

Reply:
Keep the arrow-site ones in, that's another of our repos that we'll switch over to `main` (see also ARROW-15694)

Reply:
I've included the `arrow-site` related changes and updated the description. Thank you for your help, @nealrichardson !